src/ripples/touch-ripple.jsx

```diff
@@ -74,16 +75,16 @@ const TouchRipple = React.createClass({
   },
 
   start(e, isRippleTouchGenerated) {
-    let ripples = this.state.ripples;
-
-    //Do nothing if we're starting a click-event-generated ripple
-    //while having touch-generated ripples
-    if (!isRippleTouchGenerated) {
-      for (let i = 0; i < ripples.length; i++) {
-        if (ripples[i].props.touchGenerated) return;
-      }
+    //Touch start produces a mouse down event for compat reasons. To avoid
+    //showing ripples twice we skip showing a ripple for the first mouse down
+    //after a touch start.
+    if (this.state.ignoreMouseDown && !isRippleTouchGenerated) {
+      this.setState({ ignoreMouseDown: false });
```

Comment:
Using `setState` will trigger a rerendering.
I think that it would be better to have this `ignoreMouseDown` property outside of the state.
I would just use the react state for property that have an effect on the virtual dom we render.